### ISSUE TITLE
Training curves of Hacochen code

### ISSUE THREAD

**Issue by rsundar96** [rsundar96/curriculum-learning-acceleration#1]
Following are the training curves obtained when running the [Hacochen](https://github.com/GuyHacohen/curriculum_learning) Curriculum Learning code on both CIFAR-10 and CIFAR-100-Subset_16 (small mammals superclass), which is the subset of the CIFAR-100 dataset that Hacochen used to depict the results in his paper.

#### Details on code execution:
- Learning Rate: Starts at 0.035 and gradually decreases till 0.0001
- Batch Size: 100 (*default*)
- Epochs: 140 (*default*)
- Optimiser: SGD (*default*)
- Loss Function: Categorical Cross Entropy (*default*)

### I. CIFAR-100-Subset_16 (Small Mammals superclass)

<img width="622" alt="Screenshot 2020-07-06 at 3 30 09 AM" src="https://user-images.githubusercontent.com/19652780/86584252-836c8a80-bf7c-11ea-947b-658da9888fcb.png">

**Final Training Accuracy** (Curriculum Learning): 73.0%

<img width="622" alt="Screenshot 2020-07-06 at 3 30 34 AM" src="https://user-images.githubusercontent.com/19652780/86584255-849db780-bf7c-11ea-94e1-7d2560369463.png">

**Final Training Accuracy** (Random Initialisation): 72.0%

### II. CIFAR-10

<img width="617" alt="Screenshot 2020-07-06 at 3 21 59 AM" src="https://user-images.githubusercontent.com/19652780/86584242-7fd90380-bf7c-11ea-987a-afa2736e83e8.png">

**Final Training Accuracy** (Curriculum Learning): 63.0%

<img width="617" alt="Screenshot 2020-07-06 at 3 22 19 AM" src="https://user-images.githubusercontent.com/19652780/86584248-823b5d80-bf7c-11ea-8520-168907f74c37.png">

**Final Training Accuracy** (Random Initialisation): 73.0%

#### Points to note:
I executed the Random Initialisation twice on both the CIFAR-10 and CIFAR-100-Subset_16 datasets. There is quite a noticeable variance, but I've only managed to capture the snapshot of the second run. Wasn't able to clearly make out the training curve on the CIFAR-10, which is why I decided to execute on just one superclass of CIFAR-100 so that we get a better picture.

**Comment by rsundar96** [comment]
**Update**: Plots obtained after getting rid of noise when executing on CIFAR-10 dataset.

### I. CIFAR-10

(a) Vanilla

<img width="611" alt="Screenshot 2020-07-09 at 10 44 11 PM" src="https://user-images.githubusercontent.com/19652780/87093837-bdb48100-c235-11ea-9354-6a7197da9189.png">

**Final Training Accuracy**: 48.0%
**Final Test Accuracy**: 59.3%

(b) Random initialisation

<img width="611" alt="Screenshot 2020-07-09 at 10 47 15 PM" src="https://user-images.githubusercontent.com/19652780/87094073-2865bc80-c236-11ea-9f88-639f994561e4.png">

**Final Training Accuracy**: 52.0%
**Final Test Accuracy:** 59.8%

(c) Curriculum Learning (without shifting *normalise_dataset()* )

<img width="611" alt="Screenshot 2020-07-09 at 10 48 44 PM" src="https://user-images.githubusercontent.com/19652780/87094179-5cd97880-c236-11ea-9183-46eda903fc4b.png">

**Final Training Accuracy**: 62.0%
**Final Test Accuracy**: 62.3%

(d) Curriculum Learning (shifting *normalise_dataset()* )

<img width="611" alt="Screenshot 2020-07-09 at 10 50 44 PM" src="https://user-images.githubusercontent.com/19652780/87094341-a4600480-c236-11ea-970c-c88ec3a4db65.png">

**Final Training Accuracy**: 59.0%
**Final Test Accuracy**: 61.4%

#### Points to note
Doesn't seem to be *too* much of a difference in terms of final accuracies when shifting the normalisation of data before loading and balancing the order of the dataset.

**Comment by rsundar96** [comment]
Closing this as further work on this being done at Issue #3
